src/internal/connector/onedrive/collection.go

```diff
@@ -122,6 +135,26 @@ func NewCollection(
 	return c
 }
 
+// Figures out the state of a collection using prev and current path
+// This is similar to the one in exchange, but instead of checking the
+// equivalence of folders, it checks it for the actual item.
+// TODO(meain): Why is exchange checking folders?
```

Comment:
I think maybe this comment needs updated? State should be associated with folder paths (or whatever the equivalent for folders is for a given external service). Individual item deletion/creation is controlled by adding `data.Stream` instances to a collection